src/App.tsx

```diff
@@ -1,14 +1,99 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useState, useEffect, useMemo } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos, getUser } from './api';
+import { Todo } from './types/Todo';
+import { User } from './types/User';
 
 export const App: React.FC = () => {
+  const [visibleTodos, setVisibleTodos] = useState<Todo[]>([]);
+  const [selectedFilter, setSelectedFilter] = useState('all');
+  const [query, setQuery] = useState('');
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [selectedTodoId, setSelectedTodoId] = useState<number | null>(null);
+  const [user, setUser] = useState<User | null>(null);
+
+  const handleSelect = (todo: Todo) => {
+    setSelectedTodo(todo);
+    setSelectedTodoId(todo.id);
+  };
+
+  const clearSelectedTodo = () => {
+    setSelectedTodo(null);
+    setSelectedTodoId(null);
+  };
+
+  const handleChange = (
+    event: React.ChangeEvent<HTMLSelectElement>,
+  ) => {
+    setSelectedFilter(event.target.value);
+  };
+
+  const handleSearch = (
+    event: React.ChangeEvent<HTMLInputElement>,
+  ) => {
+    setQuery(event.target.value);
+  };
+
+  const clearQuery = () => {
+    setQuery('');
+  };
+
+  useEffect(() => {
+    const todosFromServer = async () => {
+      const newTodos: Todo[] = await getTodos();
+
+      if (selectedFilter === 'all') {
+        setVisibleTodos(newTodos);
+      }
+
+      if (selectedFilter === 'completed') {
+        setVisibleTodos(newTodos.filter(todo => todo.completed === true));
+      }
+
+      if (selectedFilter === 'active') {
+        setVisibleTodos(newTodos.filter(todo => todo.completed === false));
+      }
+    };
+
+    todosFromServer();
+  }, [selectedFilter]);
```

Comment:
What's the point of loading todos every time you change the filter? You load it once and then filter the loaded todos.